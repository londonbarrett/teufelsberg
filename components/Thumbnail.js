import * as React from 'react';
import PropTypes from 'prop-types';
import Router from 'next/router';
import config from '../config';

class Thumbnail extends React.Component {
  onClick = (e) => {
    const { uri, alt } = this.props;
    e.preventDefault();
    Router.push(`/photo?id=${uri}&title=${alt}`);
  }
  render() {
    const { uri, alt } = this.props;
    return (
      <a
        className="link"
        href={`/photo?id=${uri}&title=${alt}`}
        onClick={this.onClick}
      >
        <img
          src={`${config.imagePrefix}${uri}${config.smallSuffix}`}
          alt={alt}
        />
        <style jsx>
          {`
            .link {
              box-sizing: border-box;
              display: block;
              height: 136px;
              margin: 1rem;
            }
            .link:hover {
              border-bottom: 3px solid orange;
            }
          `}
        </style>
      </a>
    );
  }
}

Thumbnail.propTypes = {
  uri: PropTypes.string.isRequired,
  alt: PropTypes.string.isRequired,
};

export default Thumbnail;
