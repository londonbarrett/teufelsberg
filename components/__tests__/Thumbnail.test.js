/* eslint-env jest */

import React from 'react';
import renderer from 'react-test-renderer';
import Thumbnail from '../Thumbnail';

describe('Thumbnail Tests', () => {
  it('Renders correctly', () => {
    const component = renderer.create(<Thumbnail
      uri="image_uri"
      alt="alternative text"
    />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
