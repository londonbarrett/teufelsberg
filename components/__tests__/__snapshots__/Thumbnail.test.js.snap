// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Thumbnail Tests Renders correctly 1`] = `
<a
  className="jsx-706898208 link"
  href="/photo?id=image_uri&title=alternative text"
  onClick={[Function]}
>
  <img
    alt="alternative text"
    className="jsx-706898208"
    src="https://image_uri_2.jpg"
  />
</a>
`;
